### PR TITLE
Feature/select component

### PR DESCRIPTION
**Pull request by emilyseibert**
# PR for #200.

## PR Status
**READY**

## Ready for PR?
- [X] I have updated the CHANGELOG with an entry including a description, a link to my profile, and a link to the issue ticket. This change is filed under an Enhancement or Bug Fix, is shown under the relevant release tag name, and is included in my PR branch. 

## Changes Proposed:
- Created a Select Component that formats all input functionality
- Select Component supports inline errors
- Updated Simulator and join modal to use new component

## Screenshot of feature:
image

@emilyseibert @oconn @dehli

### CODE REVIEW

File: less/components/select-input.less [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+@select-margin-bottom: 30px;
+
+.select-input {
+    .label;
+
+    .dropdown {
+        width: 100%;
+        position: relative;
+        cursor: pointer;
+        height: 45px;
```
Comment (oconn) [inline less/components/select-input.less]: What are your thoughts on all `px` values being stored in variables? I know this was something we were doing on other projects, but don't have a super strong opinion either way. My current opinion is that it does help organize the code a bit and help point out re-usable values.  For example, this `45px` value could be stored in a `@dropdown-height` var and used for the two `line-height` values below.

File: src/cljs/wombats_web_client/components/select-input.cljs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+(defn select-input-on-focus [form-state error-key]
+  (swap! form-state assoc error-key nil))
+
+(defn select-input 
+  [params]
+  (let [cmpnt-state (reagent/atom {:is-closed true
+                                   :selected-option nil})]
+    (fn [params]
+      (let [{:keys [form-state
```
Comment (oconn) [inline src/cljs/wombats_web_client/components/select-input.cljs]: What is the thought behind sending the entire form state to the component?

File: src/cljs/wombats_web_client/utils/forms.cljs [hunk 3]
```diff
@@ -4,6 +4,10 @@
 (defn get-value [element]
   (-> element .-target .-value))
 
+(defn optionize [id-key name-key coll]
```
Comment (oconn) [inline src/cljs/wombats_web_client/utils/forms.cljs]: awesome name 😄

File: src/cljs/wombats_web_client/components/select-input.cljs (hunk 2 above)
Comment (emilyseibert) [inline src/cljs/wombats_web_client/components/select-input.cljs]: form state gives me access to updating the form state with the selected value and also passes an inline error down to the component. in general, the inline error could be a server error, a form submission validation error, or a validate error from the component itself. storing value state in the form allows me to pass the form inline errors

File: less/components/select-input.less (hunk 1 above)
Comment (emilyseibert) [inline less/components/select-input.less]: if the `px` defined is dependent on another `px` in the file, then it should be defined. In this case, the line height and height need to be the same value so I'll create a variable for `45px`

File: src/cljs/wombats_web_client/components/select-input.cljs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    [:li.option {:key key
+                 :on-click #(option-on-click form-state form-key cmpnt-state option)}
+     name]))
+
+(defn select-input-on-click [cmpnt-state]
+  (let [new-closed-state (not (:is-closed @cmpnt-state))]
+    (swap! cmpnt-state assoc :is-closed new-closed-state)))
+
+(defn select-input-on-blur [cmpnt-state form-state error-key]
+  (swap! cmpnt-state assoc :is-closed true)
```
Comment (dehli) [inline src/cljs/wombats_web_client/components/select-input.cljs]: Can you accomplish any of this blur, focus logic using the :focus pseudo class?

**Pull request merged**